### PR TITLE
Enhance documention in `define.R`

### LINKED ISSUE #302: Improve documentation for `define_enroll_rate()`
Suggestions from Keaven:

> Before we had a prevalence for each stratum SEPARATE from enrollment rates. This assumed there would be a set of overall piecewise enrollment rates and then a multinomial probability would be used to derive stratum randomly for each observation. The help file does not document behavior well.
>
> In particular, it is not clear what the 2nd example in the help file for `define_enroll_rate()` even means.

(I wonder if `define_fail_rate()` also needs similar attention)

### PR DESCRIPTION
**Pull request by elong0527**
Fixes #302 

- provide function description.
- update example in `define_fail_rate`.

### CODE REVIEW

File: R/define.R
Comment (elong0527) [reply R/define.R:21]: The current define and AHR functions did not consider different prevalence in each stratum. From the source code of the AHR function. It seems only work for equal weight. Feel a different issue is required to address unequal prevalence in each stratum before we can document the details 

The 2nd example has been updated, I feel the example is self explained.

**Comment by LittleBeannie** [comment]
Here is an example of stratified design: https://merck.github.io/gsDesign2/articles/story-ahr-under-nph.html#design-scenario-1.

File: R/define.R
Comment (nanxstats) [inline R/define.R:21]: 1. define -> define**s**
2. Avoid starting the sentence with obvious stuff like "This function..." or "This package...".

File: R/define.R
Comment (nanxstats) [inline R/define.R:22]: stratified design**s** that have different enrollment rate**s**

File: R/define.R
Comment (nanxstats) [inline R/define.R:83]: 1. define**s**
2. "This function"

File: R/define.R
Comment (nanxstats) [inline R/define.R:84]: stratified design**s** that have different enrollment rate**s**

File: R/define.R
Comment (nanxstats) [inline R/define.R:91]: gsDesign and gsDesign2 both used "treatment group" instead of "case group" consistently. There was not a single "case group" usage.

File: R/define.R
Comment (nanxstats) [inline R/define.R:107]: For both examples, please please consider explaining in plain language what are being defined - this is because something looks natural to you doesn't mean it's "self-explanatory" to all - maybe it's because you are more familiar with the spec. 🙏

**Review by nanxstats — CHANGES_REQUESTED** [review]
I made a few comments. Please consider patching them up. I agree with @LittleBeannie that some aspects of the spec deserve better articulation.

File: R/define.R
Comment (elong0527) [reply R/define.R:118]: The doc are updated. Please take another look to see if it is clear.

**Review by LittleBeannie — APPROVED** [review]
Thanks for the updates, Yilong! I just approved this PR and will merge it after Nan's approval.

**Review by nanxstats — APPROVED** [review]
I appreciate the updates - I also improved a few details. Let's merge this.

**Pull request merged**
